### PR TITLE
Password Validator HW

### PR DESCRIPTION
**Pull request by KevinSalina**
1. Identify the component of this assignment that you did
the best on, and why. 
- Googling was a big part of this HW assignment. Especially how to search for special characters and lower/upper case without Regex. Being able to find examples online and modify the code so it works for your own application 

2.Describe in plain English what problem this assignment
was posing, and what your solution was to it.
- The main problem was how to work around Regex, in order to check a password for certain criteria.  Involved writing out own functions to test for each criteria using loops and various JS methods.

3. What was your approach or process?  
- Process was to check a main function that confirmed is all criteria returned true. From there creating a separate function to check for each criteria individually. This can make it easier to find problems if one criteria check function is not working properly.

4. What were your challenges?  
- Not being able to use regex. Understanding how to search for lower/Upper case letters.

5. What resources did you use?   
- Stack Overflow, MDN, google

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  return false
+}
+
+// Check if one character is numeric
+const checkNumeric = password => {
+  const numerics = '0123456789'
+  // let isOneNumeric = null
+
+  for (let i = 0; i < password.length; i++) {
+    if (numerics.indexOf(password[i]) >= 0) {
```
Comment (kdsbatra) [inline index.js]: Minor... but

See if you can limit the number of loops in the validators. right now you have 3 for loops effectively.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      if (chr === chr.toLowerCase() && chr !== chr.toUpperCase()) {
+        isOneLowerCase = true
+      } else if (chr === chr.toUpperCase() && chr !== chr.toLowerCase()) {
+        isOneUpperCase = true
+      }
+    }
+    i++
+  }
+  if (isOneLowerCase && isOneUpperCase) return true
+
```
Comment (kdsbatra) [inline index.js:27]: Minor:

You can return "(isOneLowerCase && isOneUpperCase)"

Instead of two different returns here.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const checkCase = password => {
+  let i = 0
+  let chr = null
+  let isOneLowerCase = null
+  let isOneUpperCase = null
+
+  while (i <= password.length) {
+    chr = password.charAt(i)
+    // Check to make sure current character is not a number
+    if (isNaN(chr * 1)) {
```
Comment (kdsbatra) [inline index.js]: Minor: but You can combine the number check with the "numeric" checks below

**Comment by kdsbatra** [comment]
Good you separated each check by function. Curious if you can do the same but use one loop only instead of 3 different loops.

Minor so it's not required.

**Comment by KevinSalina** [comment]
Trying right now to refactor down!

File: index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// One character is upper case
+const checkUpperCase = chr => {
+  return (chr === chr.toUpperCase() && chr !== chr.toLowerCase())
+}
+
+// One character is numeric
+const checkNumeric = chr => {
+  const numerics = '0123456789'
```
Comment (kdsbatra) [inline index.js:35]: why define a new const each time function runs? same for special Chars

File: index.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return (isMinLength && isLowerCase && isUpperCase && isNumeric && isSpecial)
+}
+
+// Criteria Functions
+// Min Length is 8
+const checkMinLength = password => password.length >= 8
+
+// One character is lower case
+const checkLowerCase = chr => {
+  return (chr === chr.toLowerCase() && chr !== chr.toUpperCase())
```
Comment (kdsbatra) [inline index.js:25]: chr !== chr.toUpperCase()    Am guessing you are using this for number and special chars right because for them lower and upper case is the same.

i would have prefer to perhaps use else if statements but this is fine too

File: index.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+// Main Validate Password Function
+const validatePassword = password => {
+  let isMinLength = checkMinLength(password)
+  let isLowerCase = false
+  let isUpperCase = false
+  let isNumeric = false
+  let isSpecial = false
+
+  for (let i = 0; i < password.length; i++) {
+    !isLowerCase ? isLowerCase = checkLowerCase(password[i]) : isLowerCase = true
```
Comment (kdsbatra) [inline index.js:10]: not the cleanest way to go about it... 

1. can use if statements
2. isLowerCase = !isLowerCase ? checkLowerCase(password[i]) : true
3. isLowerCase = isLowerCase || checkLowerCase(isLowerCase) ||  false
